_posts/en/newsletters/2019-02-26-newsletter.md

```diff
@@ -0,0 +1,156 @@
+    developers to build the tools necessary to put Schnorr-based systems
+    into production.  It is hoped that the code will receive additional
+    review and be ported into the upstream [libsecp256k1][] library for
+    eventual use in Bitcoin Core related to a soft fork proposal.  To
+    learn more, read the [blog post][schnorr libsecp256k1-zkp] or
+    [developer documentation][schnorr docs].
+
+## Selected Q&A from Bitcoin StackExchange
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Why does BIP44 have internal and external addresses?]({{bse}}84594)
+  External addresses are those you give to other people so that they can
+  pay you; internal addresses are those you include in your own
+  transactions for receiving change.  Pieter Wuille explains that
+  [BIP32][], upon which [BIP44][] is based, encourages using separate
+  derivation paths for these keys in case you need to prove to an
+  auditor how much money you've received but not how much money you've
+  spent (or have left).  By giving the auditor the extended public key
+  (xpub) for just the external addresses, he can track your received
+  payments but still not receive any direct information about your
+  spending or current balance via the change addresses.
+
+- [Taproot and scriptless scripts both use Schnorr, but how are they
+  different?]({{bse}}84086) In separate answers, Gregory Maxwell and
+  Andrew Chow each describe the differences between these two proposed
+  uses of Schnorr-based signatures.  Also includes a description of
+  adapter signatures, which can be used to enhance the efficiency and
+  privacy of trustless contract protocols.
+
+- [How much of block propagation time is used in verification?]({{bse}}84045)
+  Gregory Maxwell explains that it's probably closer to 0% than 1% for
+  the normal case, but that it can be much larger for a worst case block
+  that was specifically constructed to take a long time to verify.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].*
+
+- [Bitcoin Core #15348][] adds a [productivity hints][] document
+  describing tools and techniques developers have found to improve their
+  efficiency.  Although some are specific to Bitcoin Core and C++
+  development, others more generally apply to anyone who develops with
+  git or GitHub.
+
+- [LND #2419][] creates a time-based buffer for certain objects stored
+  in memory that allows them to be freed after a certain period of
+  non-use.  Previously, some unused objects were being kept in memory
+  indefinitely, leading to user-reported crashes when the program
+  exceeded the system's free memory.
```

Comment:
This seems to vague to be particularly interesting to readers. The underlying issue could potentially be a DoS vector, so I'd suggest either:

- update this summary to say that this closes a potential DoS vector where an adversary could asymmetrically exhaust node memory.
- remove it entirely.

Reply:
I especially appreciated this criticism.  I was told this solved a significant number of user-reported problems, but I couldn't see how exactly it did that, so I was indeed vague.  It turned out that there was a second half to the PR that was just merged today.  I'm deleting this item for this week and will reconsider it (with improvements) for next week after I've reviewed the second PR.  Thanks for calling me out on my vague hand waving!